activestorage/test/models/preview_test.rb

```diff
@@ -70,6 +70,61 @@ class ActiveStorage::PreviewTest < ActiveSupport::TestCase
     assert_changes -> { !!preview.image.variant(transformations)&.send(:processed?) }, to: true do
       preview.processed
     end
+
+    image = read_image(preview.image.variant(transformations))
+    assert_equal 216, image.width
+    assert_equal 280, image.height
+  end
+
+  test "#processed does not generate a variant when provided an empty transformation" do
+    blob = create_file_blob(filename: "report.pdf", content_type: "application/pdf")
+    empty_transformation = {}
+    preview = blob.preview(empty_transformation)
+
+    assert_no_changes -> { !!preview.image.variant(empty_transformation)&.send(:processed?) } do
+      preview.processed
+    end
+  end
+
+  test "#download does not generate a variant when provided an empty transformation" do
+    blob = create_file_blob(filename: "report.pdf", content_type: "application/pdf")
+    empty_transformation = {}
+    preview = blob.preview(empty_transformation)
+
+    assert_no_changes -> { !!preview.image.variant(empty_transformation)&.send(:processed?) } do
+      preview.processed.download {  }
+    end
+  end
+
+  test "#key does not generate a variant when provided an empty transformation" do
+    blob = create_file_blob(filename: "report.pdf", content_type: "application/pdf")
+    empty_transformation = {}
+    preview = blob.preview(empty_transformation)
+
+    assert_no_changes -> { !!preview.image.variant(empty_transformation)&.send(:processed?) } do
+      preview.processed.key
+    end
+  end
+
+  test "#key #download and #url raise `UnprocessedError` when preview is not processed" do
+    blob = create_file_blob(filename: "report.pdf", content_type: "application/pdf")
+    preview = blob.preview({})
```

Comment:
Since this behavior isn't specific to empty transformations, and specifying non-empty transformations is the common case, let's do that here:

```suggestion
  test "methods raise UnprocessedError when preview is not processed" do
    blob = create_file_blob(filename: "report.pdf", content_type: "application/pdf")
    preview = blob.preview(resize_to_limit: [640, 280])
```

Reply:
`methods raise ...` I find the "methods" very generic, I'm going to suggest a slight refactoring, and let me know what you think.

I agree with the non empty transformations 👍🏼